presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -389,10 +388,15 @@ public boolean catalogExists(Session session, String catalogName)
         ConnectorTransactionHandle transaction = catalogMetadata.getTransactionHandleFor(connectorId);
         ConnectorSession connectorSession = session.toConnectorSession(connectorId);
         List<ConnectorTableLayoutResult> layouts = metadata.getTableLayouts(connectorSession, connectorTable, constraint, desiredColumns);
+        if (layouts.isEmpty()) {
+            return Optional.empty();
+        }
+
+        if (layouts.size() > 1) {
```

Comment:
Why don't we change the `ConnectorMetadata#getTableLayouts` to `ConnectorMetadata#getTableLayout`  that returns a single `ConnectorTableLayoutResult`?